### ISSUE TITLE
make warnings for unused stuff

### ISSUE THREAD

**Issue by karianpour** [furstenheim/is_jsonb_valid#8]
as I ran `make install` I get the following warnings :
```
is_jsonb_valid.c: In function ‘validate_unique_items’:
is_jsonb_valid.c:764:32: warning: variable ‘r2’ set but not used [-Wunused-but-set-variable]
             JsonbIteratorToken r2;
                                ^
is_jsonb_valid.c: In function ‘validate_num_properties’:
is_jsonb_valid.c:1000:24: warning: variable ‘r’ set but not used [-Wunused-but-set-variable]
     JsonbIteratorToken r;
                        ^
is_jsonb_valid.c: In function ‘validate_num_items’:
is_jsonb_valid.c:1031:24: warning: variable ‘r’ set but not used [-Wunused-but-set-variable]
     JsonbIteratorToken r;
                        ^
is_jsonb_valid.c: In function ‘_is_jsonb_valid’:
is_jsonb_valid.c:909:44: warning: ‘refSchemaJbv’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     refSchemaJb = npath == 1 ? root_schema : JsonbValueToJsonb(refSchemaJbv);
                                            ^
is_jsonb_valid.c:792:18: note: ‘refSchemaJbv’ was declared here
     JsonbValue * refSchemaJbv;
                  ^
is_jsonb_valid.c: In function ‘validate_properties’:
is_jsonb_valid.c:449:32: warning: ‘propertiesJb’ may be used uninitialized in this function [-Wmaybe-uninitialized]
                 subSchemaJbv = findJsonbValueFromContainer(&propertiesJb->root, JB_FOBJECT, &k);
                                ^
is_jsonb_valid.c:460:42: warning: ‘additionalPropertiesJb’ may be used uninitialized in this function [-Wmaybe-uninitialized]
                     isValid = isValid && _is_jsonb_valid(additionalPropertiesJb, subDataJbv, root_schema);
                                          ^
```

**Comment by furstenheim** [comment]
Hi @karianpour thanks for your interest.
It's been a long time since I wrote the code. But if I'm not mistaken, the problem here is that the code contains asserts that are automatically removed when compiling. For example, in line 1031:

```    JsonbIteratorToken r;
   // ... 
    r = JsonbIteratorNext(&it, &v, true);
    Assert(r == WJB_BEGIN_ARRAY);
```

If I'm not mistaken, this code is executed during tests, but when compiling the final version the line with asserts is stripped. Which essentially makes "r" never to be used. If you know how to remove these particular warnings a pr is welcome.

As for the warnings with line 449. The whole context of the function is:

```if (propertiesJbv != NULL) {
                JsonbValue * subSchemaJbv;
                Jsonb * subSchemaJb;
                subSchemaJbv = findJsonbValueFromContainer(&propertiesJb->root, JB_FOBJECT, &k);
```
However, some lines before we have:

```
        if (propertiesJbv != NULL) {
                propertiesJb = JsonbValueToJsonb(propertiesJbv);
```
So basically we can only go into that branch if propertiesJb is initialized. According to [stackoverflow.com](https://stackoverflow.com/questions/14132898/gcc-wuninitialized-wmaybe-uninitialized-issues) gcc is known to raise unintialized warnings that are not bound to happen.

**Comment by furstenheim** [comment]
Hey @karianpour, I've opened a pr to fix those warnings. Although in some versions of gcc I get warnings now about unknown pragmas.
